grype/version/semantic_constraint.go

```diff
@@ -14,6 +14,7 @@ var normalizer = strings.NewReplacer(".alpha", "-alpha", ".beta", "-beta", ".rc"
 type semanticConstraint struct {
 	raw        string
 	constraint hashiVer.Constraints
+	checker    func(hashiVer.Constraints, *Version) (bool, error)
```

Comment:
this allows for others to monkey patch behavior onto the object. While this works I think there are better ways of achieving this (I think struct embedding with method overriding).